### PR TITLE
Adding Contribution guidelines

### PR DESCRIPTION
**Pull request by HGSilveri**
We were missing this part of an open-source repo, it's important to show outsiders how they should proceed and to lay out some sort of organised system. 

I kinda made some of these rules up based on what seemed reasonable to me, so I would appreciate your review and feedback.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+First of all, thank you for wanting to contribute to Pulser! We have some guidelines you should follow in order to make the contribution process as smooth as possible.
+
+## Reporting a bug or suggesting a feature
+
+The steps to take will depend on what you want to do, but generally you'll want to start by raising an issue. If you have found a bug or have a feature you would like see added to **Pulser**, you're welcome to raise an issue on [Pulser's GitHub issue tracker](https://github.com/pasqal-io/Pulser/issues). Some steps to take here:
+
+1. Do a quick search for keywords over the existing issues to ensure yours has not been added yet.
+2. If you can't find your issue already listed, create a new one. Please try to be as clear and detailed as possible in your description.
+- If you just want to give a suggestion or report a bug that's already excellent and we thank you for it! Your issue will be listed and, hopefully, someone will take care of it some point.
```
Comment (sebgrijalva) [inline CONTRIBUTING.md:11]: I suggest a comma for reading facility

```suggestion
- If you just want to give a suggestion or report a bug, that's already excellent and we thank you for it! Your issue will be listed and, hopefully, someone will take care of it some point. 
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+First of all, thank you for wanting to contribute to Pulser! We have some guidelines you should follow in order to make the contribution process as smooth as possible.
+
+## Reporting a bug or suggesting a feature
+
+The steps to take will depend on what you want to do, but generally you'll want to start by raising an issue. If you have found a bug or have a feature you would like see added to **Pulser**, you're welcome to raise an issue on [Pulser's GitHub issue tracker](https://github.com/pasqal-io/Pulser/issues). Some steps to take here:
+
+1. Do a quick search for keywords over the existing issues to ensure yours has not been added yet.
+2. If you can't find your issue already listed, create a new one. Please try to be as clear and detailed as possible in your description.
+- If you just want to give a suggestion or report a bug that's already excellent and we thank you for it! Your issue will be listed and, hopefully, someone will take care of it some point. 
+- However, you may also want to be the one solving your issue, which would be even better! In these cases, you would proceed to preparing a [Pull Request](#making-a-pull-request).
```
Comment (sebgrijalva) [inline CONTRIBUTING.md:12]: Perhaps "proceed to prepare" would work too

```suggestion
- However, you may also want to be the one solving your issue, which would be even better! In these cases, you would proceed by preparing a [Pull Request](#making-a-pull-request).
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    ```bash
+    git clone https://github.com/USERNAME/Pulser.git
+    ```
+    **Note**: `USERNAME` should be replaced by your own GitHub ID.
+
+1. Have the related issue assigned to you. We suggest that you work only on issues that have been assigned to you; by doing this, you make sure to be the only one working on this and we prevent everyone from doing duplicate work. If a related issue does not exist yet, consult the [section above](#reporting-a-bug-or-suggesting-a-feature) to see how to proceed. 
+
+2. You'll want to create a new branch where you will do your changes. **Do not push changes to your fork's `master` branch**, it should be used only to keep your fork in sync with Pulser's `master` branch (more on how to do this later). Go to the location where you cloned your fork and do:
+    ```bash
```
Comment (sebgrijalva) [inline CONTRIBUTING.md:28]: Just a comment: THANKS, this is something that is seldom said in guidelines.

**Review by sebgrijalva — DISMISSED** [review]
This looks fine, I just added some suggestions for the text.

File: CONTRIBUTING.md (hunk 3 above)
Comment (HGSilveri) [reply CONTRIBUTING.md:28]: I've had my share of issues with this too.

**Review by sebgrijalva — APPROVED** [review]

**Pull request merged**